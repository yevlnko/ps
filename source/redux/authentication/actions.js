// Instruments
import { types } from "./types";

export const authActions = Object.freeze({
    login: (credentials) => ({
        type:    types.LOGIN,
        payload: credentials,
    }),
    loginSuccess: () => ({
        type: types.LOGIN_SUCCESS,
    }),
    loginFail: (error) => ({
        type:    types.LOGIN_FAIL,
        payload: error,
        error:   true,
    }),

    authenticate: (token) => ({
        type:    types.AUTHENTICATE,
        payload: token,
    }),
    authenticateSuccess: () => ({
        type: types.AUTHENTICATE_SUCCESS,
    }),
    authenticateFail: (error) => ({
        type:    types.AUTHENTICATE_FAIL,
        payload: error,
        error:   true,
    }),

    signup: (credentials) => ({
        type:    types.SIGNUP,
        payload: credentials,
    }),
    signupSuccess: () => ({
        type: types.SIGNUP_SUCCESS,
    }),
    signupFail: (error) => ({
        type:    types.SIGNUP_FAIL,
        payload: error,
        error:   true,
    }),

    logout: () => ({
        type: types.LOGOUT,
    }),
    logoutSuccess: () => ({
        type: types.LOGOUT_SUCCESS,
    }),
    logoutFail: (error) => ({
        type:    types.LOGOUT_FAIL,
        payload: error,
        error:   true,
    }),
});
