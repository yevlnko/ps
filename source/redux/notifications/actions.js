// Instruments
import { types } from "./types";
import { v4 } from "uuid";

export const notificationsActions = Object.freeze({
    showNotification: (error) => ({
        type:    types.SHOW_NOTIFICATION,
        payload: {
            id: v4(),
            error,
        },
    }),
    hideNotification: (id) => ({
        type:    types.HIDE_NOTIFICATION,
        payload: id,
    }),
});
