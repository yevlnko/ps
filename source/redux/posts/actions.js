// Instruments
import { types } from "./types";

export const postsActions = Object.freeze({
    fetchPosts: () => ({
        type: types.FETCH_POSTS,
    }),
    fetchPostsFail: (error) => ({
        type:    types.FETCH_POSTS_FAIL,
        payload: error,
        error:   true,
    }),
    fetchPostsSuccess: (posts) => ({
        type:    types.FETCH_POSTS_SUCCESS,
        payload: posts,
    }),
    createPost: (post) => ({
        type:    types.CREATE_POSTS,
        payload: post,
    }),
    removePost: (id) => ({
        type:    types.REMOVE_POSTS,
        payload: id,
    }),
    likePost: (data) => ({
        type:    types.LIKE_POST,
        payload: data,
    }),
    unlikePost: (data) => ({
        type:    types.UNLIKE_POST,
        payload: data,
    }),
});
