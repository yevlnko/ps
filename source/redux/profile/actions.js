// Instruments
import { types } from "./types";

export const profileActions = Object.freeze({
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),
    updateAvatar: (avatar) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatar,
    }),
});
