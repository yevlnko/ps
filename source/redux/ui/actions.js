// Instruments
import { types } from "./types";

export const uiActions = Object.freeze({
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    initializeSuccess: () => ({
        type: types.INITIALIZE_SUCCESS,
    }),
    setOnlineState: (onlineState) => ({
        type:    types.SET_ONLINE_STATE,
        payload: onlineState,
    }),
    setPostFetchingState: (postFetchingState) => ({
        type:    types.SET_POSTS_FETCHING_STATE,
        payload: postFetchingState,
    }),
    setAuthFetchingState: (authFetchingState) => ({
        type:    types.SET_AUTH_FETCHING_STATE,
        payload: authFetchingState,
    }),
    emmitError: (error, metaMessage = null) => ({
        type:    types.EMMIT_ERROR,
        payload: error,
        error:   true,
        meta:    metaMessage,
    }),
    setProfileFetchingState: (state) => ({
        type:    types.SET_PROFILE_FETCHING_STATE,
        payload: state,
    }),
    setProfileEditingState: (state) => ({
        type:    types.SET_PROFILE_EDITING_STATE,
        payload: state,
    }),
    setAvatarFetchingState: (state) => ({
        type:    types.SET_AVATAR_FETCHING_STATE,
        payload: state,
    }),
    setPasswordEditingState: (state) => ({
        type:    types.SET_PASSWORD_EDITING_STATE,
        payload: state,
    }),
});
