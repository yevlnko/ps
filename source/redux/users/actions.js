// Instruments
import { types } from "./types";

export const usersActions = Object.freeze({
    fillUsers: (users) => ({
        type:    types.FILL_USERS,
        payload: users,
    }),
    clearUsers: () => ({
        type: types.CLEAR_USERS,
    }),
});
