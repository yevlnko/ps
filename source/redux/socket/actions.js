// Instruments
import { socket } from "init/socket";
import { uiActions } from "redux/ui/actions";
import { postsActions } from "redux/posts/actions";


export const socketActions = Object.freeze({
    listenConnection: () => (dispatch) => {
        socket.on('connect', () => {
            dispatch(uiActions.setOnlineState(true));
        });
        socket.on('disconnect', () => {
            dispatch(uiActions.setOnlineState(false));
        });
    },
    listenPosts: () => (dispatch, getState) => {

        const currentUserID = getState().profile.get('id');

        socket.on('create', (response) => {
            const { data: post, meta } = JSON.parse(response);

            if (currentUserID === meta.userID) {
                return null;
            }

            dispatch(postsActions.createPost(post));
        });

        socket.on('remove', (response) => {
            const { data: postId, meta } = JSON.parse(response);

            if (currentUserID === meta.userID) {
                return null;
            }

            dispatch(postsActions.removePost(postId));
        });


        socket.on('like', (response) => {
            const {
                data: { userID, postID },
                meta,
            } = JSON.parse(response);

            if (currentUserID === meta.userID) {
                return null;
            }

            if (meta.action === 'like') {
                const liker = getState()
                    .users.find((user) => user.get('id') === userID)
                    .delete('avatar');

                dispatch(
                    postsActions.likePost({
                        id: postID,
                        liker,
                    })
                );

                return null;
            }

            dispatch(postsActions.unlikePost({
                id:     postID,
                userId: userID,
            }));

        });

    },
});
