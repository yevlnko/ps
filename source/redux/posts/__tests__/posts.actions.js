import { postsActions } from "../actions";

// GLOBAL VARIABLE
// __.credentials  from setup JEST FILE
// __.error from setup JEST FILE

const posts =  [
    {
        id:      '5b163e507e057d6e2cecf203',
        comment: 'Nothing',
        likes:   [
            {
                id:        '5b140df97e057d69eaecf021',
                firstName: 'Александр',
                lastName:  'Коник',
            }
        ],
        created: 1528184400,
        author:  {
            id:        '5b16369b7e057d0d90ecf1fe',
            firstName: 'Jubei',
            lastName:  'Kibagami',
            avatar:    'https://lab.lectrum.io/redux/api/image/zgthxt3betpy/LX9h3dCJ7W.jpg',
        },
    }];


describe('Posts actions:', () => {

    test('FETCH_POSTS', () => {
        expect(postsActions.fetchPosts()).toMatchSnapshot();
    });
    test('FETCH_POSTS_FAIL', () => {
        expect(postsActions.fetchPostsFail()).toMatchSnapshot();
    });
    test('FETCH_POSTS_SUCCESS', () => {
        expect(postsActions.fetchPostsSuccess(posts)).toMatchSnapshot();
    });
    test('CREATE_POSTS', () => {
        expect(postsActions.createPost()).toMatchSnapshot();
    });
    test('REMOVE_POSTS', () => {
        expect(postsActions.removePost()).toMatchSnapshot();
    });
    test('LIKE_POST', () => {
        expect(postsActions.likePost()).toMatchSnapshot();
    });
    test('UNLIKE_POST', () => {
        expect(postsActions.unlikePost()).toMatchSnapshot();
    });

});
