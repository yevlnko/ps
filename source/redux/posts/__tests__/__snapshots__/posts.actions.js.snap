// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts actions: CREATE_POSTS 1`] = `
Object {
  "payload": undefined,
  "type": "CREATE_POSTS",
}
`;

exports[`Posts actions: FETCH_POSTS 1`] = `
Object {
  "type": "FETCH_POSTS",
}
`;

exports[`Posts actions: FETCH_POSTS_FAIL 1`] = `
Object {
  "error": true,
  "payload": undefined,
  "type": "FETCH_POSTS_FAIL",
}
`;

exports[`Posts actions: FETCH_POSTS_SUCCESS 1`] = `
Object {
  "payload": Array [
    Object {
      "author": Object {
        "avatar": "https://lab.lectrum.io/redux/api/image/zgthxt3betpy/LX9h3dCJ7W.jpg",
        "firstName": "Jubei",
        "id": "5b16369b7e057d0d90ecf1fe",
        "lastName": "Kibagami",
      },
      "comment": "Nothing",
      "created": 1528184400,
      "id": "5b163e507e057d6e2cecf203",
      "likes": Array [
        Object {
          "firstName": "Александр",
          "id": "5b140df97e057d69eaecf021",
          "lastName": "Коник",
        },
      ],
    },
  ],
  "type": "FETCH_POSTS_SUCCESS",
}
`;

exports[`Posts actions: LIKE_POST 1`] = `
Object {
  "payload": undefined,
  "type": "LIKE_POST",
}
`;

exports[`Posts actions: REMOVE_POSTS 1`] = `
Object {
  "payload": undefined,
  "type": "REMOVE_POSTS",
}
`;

exports[`Posts actions: UNLIKE_POST 1`] = `
Object {
  "payload": undefined,
  "type": "UNLIKE_POST",
}
`;
